lib/ldclient-rb/impl/util.rb

```diff
@@ -75,6 +75,21 @@ def self.validate_application_info(app, logger)
           version: validate_application_value(app[:version], :version, logger),
         }
       end
+
+      #
+      # @param value [String, nil]
+      # @param logger [Logger]
+      # @return [String, nil]
+      #
+      def self.validate_payload_filter_key(value, logger)
+        return nil if value.nil?
+        return value if value.is_a?(String) && /^[a-zA-Z0-9][a-zA-Z0-9._-]*$/.match?(value)
```

Comment:
Your version is better because we don't have that awkward `\` escape, but I think it should match this one exactly:
```
^[a-zA-Z0-9][._\-a-zA-Z0-9]*$
```
otherwise we'll now have 3 different versions in the wild

Reply:
Done!